tomplot/local_area_plot.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+
+
+def area_restriction(field_data, coords_X, coords_Y, Y_lim=None, X_lim=None):
```

Comment:
I think this is a good name for your function.

We could make it a bit more general, so this won't be relevant in the end, but we should have the `X_lim` and `Y_lim` arguments swapped (to match the same order as `coords_X` and `coords_Y`).

See below for my suggestion about this!